library_generation/new_client/new-client.py

```diff
@@ -0,0 +1,458 @@
+         "Maven Central"
+)
+@click.option(
+    "--release-level",
+    type=click.Choice(["stable", "preview"]),
+    default="preview",
+    show_default=True,
+    help="A label that appears in repo-metadata.json. The first library "
+         "generation is always 'preview'."
+)
+@click.option(
+    "--transport",
+    type=click.Choice(["grpc", "http", "both"]),
+    default="grpc",
+    show_default=True,
+    help="A label that appears in repo-metadata.json"
+)
+@click.option("--language", type=str, default="java", show_default=True)
+@click.option(
+    "--distribution-name",
+    type=str,
+    help="Maven coordinates of the generated library. By default it's "
+         "com.google.cloud:google-cloud-<api_shortname>"
+)
+@click.option(
+    "--api-id",
+    type=str,
+    help="The value of the apiid parameter used in README.md It has link to "
+         "https://console.cloud.google.com/flows/enableapi?apiid=<api_id>"
+)
+@click.option(
+    "--requires-billing",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="Based on this value, README.md explains whether billing setup is "
+         "needed or not."
+)
+@click.option(
+    "--proto-path",
+    required=True,
+    type=str,
+    default=None,
+    help="Path to proto file from the root of the googleapis repository to the"
+         "directory that contains the proto files (without the version)."
+         "For example, to generate the library for 'google/maps/routing/v2', "
+         "then you specify this value as 'google/maps/routing'"
+)
+@click.option(
+    "--cloud-api",
+    type=bool,
+    default=True,
+    show_default=True,
+    help="If true, the artifact ID of the library is 'google-cloud-'; "
+         "otherwise 'google-'"
+)
+@click.option(
+    "--group-id",
+    type=str,
+    default="com.google.cloud",
+    show_default=True,
+    help="The group ID of the artifact when distribution name is not set"
+)
+@click.option(
+    "--library-type",
+    type=str,
+    default="GAPIC_AUTO",
+    show_default=True,
+    help="A label that appear in repo-metadata.json to tell how the library is "
+         "maintained or generated"
+)
+@click.option(
+    "--googleapis-url",
+    type=str,
+    default="https://github.com/googleapis/googleapis.git",
+    show_default=True,
+    help="The URL of the repository that has proto service definition"
+)
+@click.option(
+    "--rest-docs",
+    type=str,
+    help="If it exists, link to the REST Documentation for a service"
+)
+@click.option(
+    "--rpc-docs",
+    type=str,
+    help="If it exists, link to the RPC Documentation for a service"
+)
+@click.option(
+    "--split-repo",
+    type=bool,
+    default=False,
+    help="Whether generating a library into a split repository"
+)
+def generate(
+    api_shortname,
+    name_pretty,
+    product_docs,
+    api_description,
+    generator_version,
+    release_level,
+    distribution_name,
+    api_id,
+    requires_billing,
+    transport,
```

Comment:
This should be removed in the future, the value of `transport` should be parsed from `Build.bazel`.